services/processor/native/compiling_repository.go

```diff
@@ -21,46 +21,67 @@ import (
 	"time"
 )
 
-func initializePreBuiltContractInstances() map[string]*types.ContractInstance {
-	res := make(map[string]*types.ContractInstance)
-	for contractName, contractInfo := range repository.PreBuiltContracts {
-		instance, err := types.NewContractInstance(contractInfo)
-		if err == nil {
-			res[contractName] = instance
+type CompositeRepository struct {
```

Comment:
i am unsure about the array ... is it not pre-mature optimisation ? we only have pre-build and deployed contracts now ...